Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -1,32 +1,50 @@
 #include "BatteryStateOfChargeService.h"
+#include "BatteryData.h"
 
 namespace
 {
     const double BATTERY_AMP_HOUR_CAPACITY = 123.0;
+    const double MS_IN_HOUR = 3600000;
 }
 
 BatteryStateOfChargeService::BatteryStateOfChargeService(double initialStateOfChargePercent)
 : initialStateOfChargePercent_(initialStateOfChargePercent)
 {
+    ampHoursUsed_ = BATTERY_AMP_HOUR_CAPACITY * initialStateOfChargePercent_ / 100;
 }
 
+// Empty since children will be destroyed
 BatteryStateOfChargeService::~BatteryStateOfChargeService()
 {
 }
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return ampHoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if (dataPoint_.current < 0) {
+       return true;
+    }
     return false;
 }
 
+// Amphours used divided by current or remaining amphours divided by current
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double hoursTillChargedOrDepleted;
+    QTime msDuration(0,0,0,0);
+
+    if (isCharging()) {
+        hoursTillChargedOrDepleted = ampHoursUsed_ / dataPoint_.current * -1;
```

Comment:
There's an absolute value method that might be more useful here, rather than multiplying by -1.